### PR TITLE
Feature/eui 1599 open id connect

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/auth/index.ts [hunk 1]
```diff
@@ -1,179 +1,163 @@
...
-            console.log('THIS USER CAN NOT LOGIN');
-            // tslint:disable-next-line
-            res.redirect(`${getConfigValue(SERVICES_IDAM_API_PATH)}/login?response_type=code&client_id=${getConfigValue(IDAM_CLIENT)}&redirect_uri=${getConfigValue(PROTOCOL)}://${req.headers.host}/oauth2/callback&scope=profile openid roles manage-user create-user manage-roles`)
-            return false
-        }
-        // set browser cookie
-        res.cookie(getConfigValue(COOKIE_TOKEN), accessToken)
+export async function doLogout(req: Request, res: Response, status = 302) {
+
+    req.query.redirect = app.locals.client.endSessionUrl({
```
Comment (paddybasi) [inline api/auth/index.ts]: look at MC demo

Comment (udaydenduluri33) [reply api/auth/index.ts]: @paddybasi  I believe this is done now.

File: api/auth/index.ts [hunk 2]
```diff
@@ -1,179 +1,163 @@
...
+
+    passport.unuse('oidc').use('oidc', new Strategy({
+        client: app.locals.client,
+        params: {
+            prompt: 'login',
+            redirect_uri: redirectUri,
+            scope: 'profile openid roles manage-user create-user',
+        },
+        sessionKey: 'xui_webapp', // being explicit here so we can set manually on logout
+        usePKCE: false, // issuer doesn't support pkce - no code_challenge_methods_supported
```
Comment (ficklephil) [inline api/auth/index.ts]: Think about using block level comments over in-line commenting, something for later.

File: config/default.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
   "idamClient": "xuiaowebapp",
   "microservice": "xui_webapp",
   "now": false,
-  "oauthCallbackUrl": "/oauth2/callback",
+  "oauthCallbackUrl": "oauth2/callback",
```
Comment (ficklephil) [inline config/default.json]: Looks good.

File: src/app/containers/app/app.component.ts [hunk 4]
```diff
@@ -14,29 +15,95 @@ import { environment as config } from '../../../environments/environment';
...
+    const route$ = this.store.pipe(select(fromRoot.getRouterUrl));
+    const userIdleSession$ =  this.store.pipe(select(fromRoot.getUserIdleTime));
+    const userTimeOut$ =  this.store.pipe(select(fromRoot.getUserTimeOut));
+    combineLatest([
+      route$.pipe(first(value => typeof value === 'string' )),
+      userIdleSession$.pipe(filter(value => !isNaN(value)), take(1)),
+      userTimeOut$.pipe(filter(value => !isNaN(value)), take(1))
+    ]).subscribe(([routes, idleMilliseconds, timeout]) => {
+      const isSignedOut: boolean = routes.indexOf('signed-out') !== -1;
+      if (timeout && idleMilliseconds && !isSignedOut) {
```
Comment (ficklephil) [inline src/app/containers/app/app.component.ts]: I would be tempted to abstract this to a separate function ( but later on )

**Comment by ficklephil** [comment]
Looks good to me! I didn't test it locally, so as long as it works locally, it's all good.

**Pull request merged**